lib/harvester_utils/downloader.rb

```diff
@@ -18,13 +20,32 @@ def self.url_exists?(url)
     end
 
     def self.download_file(url)
-      response = Faraday.get(url)
+      conn = Faraday.new(request: { timeout: 300 })
+      response = conn.get(url)
+
       raise DownloadError, "Unexpected response status: #{response.status} for url: #{url}" if response.status != 200
-      file = Tempfile.new(rand(1_000_000).to_s)
-      file.binmode
-      file.write(response.body)
-      file.close
-      file
+
+      zip_file = nil
+      Tempfile.open do |file|
+        file.binmode
+        file.write(response.body)
+        zip_file = file
+      end
+
+      self.extract_csv(zip_file)
```

Comment:
Tato zmena sa mi nepozdava, metoda povodne len stahovala subor, teraz ho stahuje a este aj hlada csv a to rozpakuje? Ak nam tu nepadaju testy tak nieco je velmi zle